src/submission-service.ts

```diff
@@ -299,6 +306,18 @@ async function executePostSubmissionAction(
   }
 }
 
+async function uploadAttachment({
+  formId,
+  file,
+}: {
+  formId: number
+  file: UploadAttachmentConfiguration
+}) {
+  const creds = await generateUploadAttachmentCredentials(formId)
+  await uploadAttachmentToS3(creds, file)
+  return creds.attachmentDataId
```

Comment:
This should be returning a lot more then just the id, ideally we return the object set on in the submission data. e.g.

```json

{
  "s3": {
    "bucket": "",
    "key": "",
    "region": ""
  },
  "url": "",
  "contentType": "",
  "fileName": "",
  "id": ""
}
```